python/lsst/pipe/tasks/insertFakeSources.py

```diff
@@ -0,0 +1,597 @@
+    `mkFakeStars`
+        Use the PSF information from the calexp to make a fake star using the magnitude information from the
+        input file.
+    `cleanCat`
+        Remove rows of the input fake catalog which have half light radius, of either the bulge or the disk,
+        that are 0.
+    `addFakeSources`
+        Add the fake sources to the calexp.
+
+    Notes
+    -----
+    The ``calexp`` with fake souces added to it is written out as the datatype ``calexp_fakes``.
+    """
+
+    _DefaultName = "insertFakeSources"
+    ConfigClass = InsertFakeSourcesConfig
+
+    def runDataRef(self, dataRef):
+        """Read in/write out the required data products and add fake sources to the calexp.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.butlerSubset.ButlerDataRef`
+            Data reference defining the ccd to have fake added to it
+            Used to access the following data products:
+                calexp
+                jointcal_wcs
+                jointcal_photoCalib
+
+        Notes
+        -----
+        Uses the calibration and WCS information attached to the calexp for the posistioning and calibration
+        of the sources unless the config option config.useUpdatedCalibs is set then it uses the
+        meas_mosaic/jointCal outputs. The config defualts for the column names in the catalog of fakes are
+        taken from the UW simulations database. Operates on one ccd at a time.
+        """
+
+        dataRef.dataId["tract"] = self.config.fakeTract
+        if self.config.fakeType == "snapshot":
+            fakeCat = dataRef.get("fakeSourceCat").toDataFrame()
+            self.calexpType = "calexp"
+            self.fakeSourceCatType = "fakeSourceCat"
+        elif self.config.fakeType == "static":
+            fakeCat = dataRef.get("deepCoadd_fakeSourceCat").toDataFrame()
+            self.calexpType = self.config.calexpType
+            # To do: DM-16254, the read and write of the fake catalogs will be changed once the new pipeline
+            # task structure for ref cats is in place.
+            self.fakeSourceCatType = "deepCoadd_fakeSourceCat"
+        else:
+            fakeCat = Table.read(self.config.fakeType).to_pandas()
+            self.calexpType = self.config.calexpType
+
+        calexp = dataRef.get(self.calexpType)
+        if self.config.useUpdatedCalibs and self.calexpType == "calexp":
+            self.log.info("Using updated calibs from meas_mosaic/jointCal")
+            wcs = dataRef.get("jointcal_wcs")
+            photoCalib = dataRef.get("jointcal_photoCalib")
+        else:
+            wcs = calexp.getWcs()
+            photoCalib = calexp.getCalib()
+
+        calexpWithFakes, fakeCat = self.run(fakeCat, calexp, wcs, photoCalib)
+
+        if self.calexpType == "calexp":
+            self.processFakeCalexp(dataRef, calexpWithFakes)
+
+        dataRef.put(calexpWithFakes, "fakes_" + self.calexpType)
```

Comment:
normally our modifies to a data type are not separated with an underscore, but are a compound word, i.e. deepCoadd. normally the underscore denotes a later derived product i.e. deepCoadd_det